visualization/app/codeCharta/ui/nodeContextMenu/nodeContextMenu.component.ts

```diff
@@ -57,29 +50,37 @@ export class NodeContextMenuController implements ShowNodeContextMenuSubscriber,
 		private blacklistService: BlacklistService
 	) {
 		"ngInject"
-		NodeContextMenuController.instance = this
 		MapColorsService.subscribe(this.$rootScope, this)
-		document.addEventListener("building-right-clicked", this.onBuildingRightClicked)
-		NodeContextMenuController.subscribeToShowNodeContextMenu(this.$rootScope, this)
-		NodeContextMenuController.subscribeToHideNodeContextMenu(this.$rootScope, this)
+		Store.store.subscribe(() => {
+			const state = Store.store.getState()
+			const rightClickedNodeData = rightClickedNodeDataSelector(state)
+			if (this.rightClickedNodeData === rightClickedNodeData) return
```

Comment:
What must happen for this condition to become true?

Reply:
`rightClickedNodeDataSelector` calculates a memorized value. If the **related** input of state doesn't change, the same memorized  value / object is returned. Otherwise a new value.